### ISSUE TITLE
Dropdown: lost focus when select item using keyboard  [also lost focus when remove selected item using keyboard]

### ISSUE THREAD

**Issue by arnodemer** [primefaces/primereact#6873]
### Describe the bug

When select dropdown's item using mouse, the focus is restored on dropdown component.
But when select item using keyboard, the focus is lost.
Also when move focus to X button of selected item and hit enter, dropdown lost focus.

### PrimeReact version

10.7.0

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

1. go to Dropdown showcase: https://primereact.org/dropdown/#filter
2. open dropdown item list
3. set focus on filter input  (in our case, dropdown has autoFocus props set)
4. use keyboard arrow up/down to pre-select and item
5. hit Enter keyboard key
=> item is selected but dropdown component is **not focused**.

6. restore focus on dropdown component with selected country item
7. hit tab to move focus on X cross button
8. hit enter
=> the item is removed from dropdown (good)  but **focus is not restored** on dropdown component.

### Expected behavior

per accessibility rules, dropdown component should be focused after item selection or after remove selected item

**Comment by melloware** [comment]
Similar to https://github.com/primefaces/primereact/issues/6803

**Comment by melloware** [comment]
Siimilar to: https://github.com/primefaces/primereact/issues/6269
